### DISCUSSION TITLE
handling new versions of axe-core

### DISCUSSION THREAD

**Discussion by juprog** [TroyWalshProf/SeleniumAxeDotnet#178] [Q&A] [answered]
This should be an easy question, or maybe a dummy question :

I was curious, how does it work when axe-core comes out with new versions? 

Is this library/project automatically updated, or as users do we need to create requests for it.... 

for example, today with the report count bug fix, I did notice that the axe-core version went up as well.   But I was wondering if this would happen automatically every time. I am only doing a proof of concept, but if I were to create a solution and have to maintain it, I was just curious how the changes in the axe-core api would be handled.

thanks,

**Comment by TroyWalshProf** [comment, marked as answer]
Dependabot automatically updates this for us.
Example:
https://github.com/TroyWalshProf/SeleniumAxeDotnet/pull/177

If you want to use a custom version of Axe core this should work:
https://troywalshprof.github.io/SeleniumAxeDotnet/#/?id=axebuilderaxebuilderwebdriver-axebuilderoptions

**Reply by juprog** [reply]
perfect ! thanks. Also good to know that we could use a custom version.

**Reply by juprog** [reply]
hi,
just as a follow-up.

It seems like I am not able to update the SeleniumAxe through nuget, in my solution.
I can see that axe-core dependencies have been upgraded in this project, but I am not able to get those updates through nuget, even though I installed it from there.

Is there a reason?

**Comment by juprog** [comment]
hi,
just as a follow-up.

It seems like I am not able to update the SeleniumAxe through nuget, in my solution.
I can see that axe-core dependencies have been upgraded in this project, but I am not able to get those updates through nuget, even though I installed it from there.

Is there a reason?

**Reply by TroyWalshProf** [reply]
What version of the .Net package are you referencing?

**Reply by juprog** [reply]
v3.0.0 since a couple months back when I last updated (and noticed the axe-core update).
I am using it in a .net standard test project (not .net core)

 For past couple weeks, I could see that the axe-core dependency was updated in the project here, but I was not sure why I could not get it from update. and I was not too sure how to download/Build and change reference in my project-if there is, I can learn it, but I thought this would come from nuget?

I can see that tonight there is a new update for 3.0.1 - which I will try most likely tomorrow/this week (it looks like it has axe-core 4.3.3 so that would upgrade from 4.3.0 )
I noticed in the pipelines a "nuget publish" step was being skipped, not sure if it was related?

thanks,

**Reply by TroyWalshProf** [reply]
Axe-core took a while because I was out on vacation.  :)
Latest package (3.0.1)  uses axe-core 4.3.3

If you want to see exactly what has been released you can review:
https://github.com/TroyWalshProf/SeleniumAxeDotnet/releases

As for the pipeline, skipped means the build was a PR/Gated build.
After the PR get's approved and merged in a merge/CI build is run.
This merge/CI build queues the push to NuGet.org.  
The push requires a manual approval because we don't necessarily want to release a new version for every single dependency update.

**Reply by juprog** [reply]
thanks! It's no problem, now that I know how it works, and what to expect :)
it also makes sense for the manual approval.
